crates/ark/src/variables/variable.rs

```diff
@@ -106,6 +113,38 @@ impl WorkspaceVariableDisplayValue {
         Self::new(String::from(""), false)
     }
 
+    fn from_language(value: SEXP) -> Self {
+        if r_inherits(value, "formula") {
+            return match Self::from_formula(value) {
+                Ok(display_value) => display_value,
+                Err(err) => Self::from_error(harp::Error::Anyhow(err)),
+            };
+        }
+
+        return Self::from_error(Error::Anyhow(anyhow!("Unexpected language object type")));
+    }
+
+    fn from_formula(value: SEXP) -> anyhow::Result<Self> {
+        // `format` for formula will return a character vector, splitting the expressions within
+        // the formula, for instance `~{x + y}` will be split into `~` and `["~{", "x", "}"]`.
+        let formatted: Vec<String> = RFunction::new("base", "format")
+            .add(value)
+            .call_in(ARK_ENVS.positron_ns)?
+            .try_into()?;
+
+        let mut display_value = String::with_capacity(MAX_DISPLAY_VALUE_LENGTH);
```

Comment:
We might want to join lines with `; ` (with trailing space) and add a space after `{` and before `}`. But the deparser might add lines in other cases too, so I guess we can't easily do that.

I would have just ellipsised after the first line to avoid this issue, but your approach allows seeing more of the contents. I'll leave it up to you to decide what is best.

Reply:
Ohh I missed that. I decided to just add an ellipsis after the first `{`: https://github.com/posit-dev/ark/pull/630/commits/c812b8ff6451a7f8a4f79aadba0869dfcef1cc23